rmw_microxrcedds_c/src/rmw_guard_condition.c

```diff
@@ -23,24 +23,33 @@ rmw_create_guard_condition(
 {
   (void)context;
 
-  rmw_guard_condition_t * rmw_guard_condition = (rmw_guard_condition_t *)rmw_allocate(
-    sizeof(rmw_guard_condition_t));
-
-  rmw_guard_condition->context = context;
-  rmw_guard_condition->implementation_identifier = rmw_get_implementation_identifier();
-  rmw_guard_condition->data = (bool *)rmw_allocate(sizeof(bool));
-
-  bool * hasTriggered = (bool *)rmw_guard_condition->data;
-  *hasTriggered = false;
-
-  return rmw_guard_condition;
+  rmw_uxrce_mempool_item_t * memory_node = get_memory(&guard_condition_memory);
+  if (!memory_node) {
+    RMW_SET_ERROR_MSG("Not available memory node");
+    return NULL;
+  }
+  rmw_uxrce_guard_condition_t * custom_guard_condition =
+    (rmw_uxrce_guard_condition_t *)memory_node->data;
+  custom_guard_condition->hasTriggered = false;
+
```

Comment:
```suggestion
custom_guard_condition->rmw_guard_condition->context = context;
custom_guard_condition->rmw_guard_condition->implementation_identifier = rmw_get_implementation_identifier();
custom_guard_condition->rmw_guard_condition->data = false;

```

Should we initialize this?